### PR TITLE
[Done] Place geoblock - additional aggregation options

### PR DESCRIPTION
**Pull request by caspervdw**
This addresses https://nelen-schuurmans.atlassian.net/browse/BACK-831

WIth the new `statistic` argument in the `Place` geoblock, users can control what happens when features overlap.

I sat down and made a nice utility function that does this 'reduction' of a stack of arrays, because it will come in very useful in https://github.com/nens/dask-geomodeling/pull/32.

A future action will be rebasing `FillNoData` on the `reduce_rasters` function. If we subclass that we get `Min`, `Max`, `ArgMin`, `ArgMax`, `Percentile`, `Median`, `Mean`, `Sum`, `Count`, `Nans`, `First`, `Last`, `Std`, `Var` blocks almost for free!

### CODE REVIEW

File: dask_geomodeling/raster/reduction.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    elif statistic == "first":
+        # populate 'out' with the first value that is not 'no data'
+        for data in stack[::-1]:
+            index = get_index(data["values"], data["no_data_value"])
+            out[index] = data["values"][index]
+    elif statistic == "count":
+        # count the number of values that are not 'no data'
+        for data in stack:
+            out += get_index(data["values"], data["no_data_value"])
+    elif statistic == "nans":
```
[the remark this answers is not in the document]
Comment (caspervdw) [reply dask_geomodeling/raster/reduction.py:82]: Counting nans is also problematic for the Place geoblock, as it removes objects ("trees") that are outside of the bbox from the stack. So "nans" doesn't make sense there.

I'll remove it from the options now.

File: dask_geomodeling/raster/reduction.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+def reduce_rasters(stack, statistic="last", shape=None, fill_value=None, dtype=None):
+    """Reduces a stack of rasters, skipping 'no data' values.
+
+    Args:
+      stack (list): a list of dicts containing "values" (ndarray)
+        and "no_data_value"
+      statistic (str): the applied statistic (no data is ignored). One of:
+        {"last", "first", "count", "nans", "sum", "mean", "min",
+        "max", "argmin", "argmax", "product", "std", "var", "p<number>"}
+      shape (tuple of number): the output shape of the array
```
[the remark this answers is not in the document]
Comment (caspervdw) [reply dask_geomodeling/raster/reduction.py]: I removed the redudant shape param and dealt with the zero-length stack inside the Place geoblock

File: dask_geomodeling/raster/reduction.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+def check_statistic(statistic):
+    if statistic not in STATISTICS:
+        percentile = parse_percentile_statistic(statistic)
+        if percentile is None:
+            raise ValueError('Unknown statistic "{}"'.format(statistic))
+
+
+def reduce_rasters(stack, statistic="last", shape=None, fill_value=None, dtype=None):
+    """Reduces a stack of rasters, skipping 'no data' values.
```
[the remark this answers is not in the document]
Comment (arjanverkerk) [reply dask_geomodeling/raster/reduction.py]: What is your idea on this? For me it is not obvious that by "reduce" you mean applying some statistical function stack-wise. If you insist on the name then at least spend some words on it in the docstring.

**Comment by arjanverkerk** [comment]
Whoah, my proposed one-liners didn't make it to the review! I must have forgotten to click a submit button somewhere... Never mind.

File: dask_geomodeling/raster/reduction.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (caspervdw) [reply dask_geomodeling/raster/reduction.py]: Reduce as in numpy.ufunc.reduce or map-reduce

"Reduces a‘s dimension by one, by applying ufunc along one axis."

https://het.as.utexas.edu/HET/Software/Numpy/reference/generated/numpy.ufunc.reduce.html

**Pull request merged**